fe-library-app/src/components/BookList/BookCard/BookCard.tsx

```diff
@@ -1,12 +1,38 @@
-import { BookItemResponse } from '../../../services/BookService'
+import { useState } from 'react'
+
+import { useNavigate } from 'react-router-dom'
+
+import { BookItemResponse, deleteBook } from '../../../services/BookService'
 import DefaultBookCover from './DefaultBookCover.png'
 import styles from './BookCard.module.css'
+import AddModifyBookModal from '../../../modals/AddModifyBookModal'
 
 interface BookProp{
     book: BookItemResponse
+    onClick: () => void,
 }
 
 const BookCard = (props: BookProp) => {
+  const navigator = useNavigate()
+  const [ showModifyModule, setShowModifyModule ] = useState(false)
+
+  const handleModifyClick = () => {
+    navigator('/add_modify/' + props.book.Id.toString())
+  }
+
+  const handleDeleteClick = () => {
+    if (confirm('Are you sure that you want to delete this book?') === true) {
+      deleteBook(props.book.Id.toString())
+        .then(() => {
+          alert('You successfully delete book ' + props.book.Title)
+          props.onClick()
```

Comment:
What is the purpose of this onClick? It doesn't tell much by the name

Reply:
Understand